Frontend/library/src/Inputs/MouseController.ts

```diff
@@ -1,212 +1,85 @@
-// Copyright Epic Games, Inc. All Rights Reserved.
-
 import { MouseButtonsMask, MouseButton } from './MouseButtons';
-import { Logger } from '@epicgames-ps/lib-pixelstreamingcommon-ue5.5';
 import { StreamMessageController } from '../UeInstanceMessage/StreamMessageController';
 import { CoordinateConverter } from '../Util/CoordinateConverter';
 import { VideoPlayer } from '../VideoPlayer/VideoPlayer';
-import { IMouseEvents } from './IMouseEvents';
-import { LockedMouseEvents } from './LockedMouseEvents';
-import { HoveringMouseEvents } from './HoveringMouseEvents';
 import type { ActiveKeys } from './InputClassesFactory';
-import { EventListenerTracker } from '../Util/EventListenerTracker';
 
 /**
- * Handles the Mouse Inputs for the document
+ * Extra types for Document and WheelEvent
  */
+declare global {
+    interface Document {
+        mozPointerLockElement: unknown;
+        mozExitPointerLock?(): void;
+    }
+
+    interface WheelEvent {
+        wheelDelta: number;
+    }
+}
+
 export class MouseController {
-    videoElementProvider: VideoPlayer;
-    toStreamerMessagesProvider: StreamMessageController;
+    videoPlayer: VideoPlayer;
+    streamMessageController: StreamMessageController;
     coordinateConverter: CoordinateConverter;
-    activeKeysProvider: ActiveKeys;
+    activeKeys: ActiveKeys;
 
-    // Utility for keeping track of event handlers and to unregister them.
-    private mouseEventListenerTracker = new EventListenerTracker();
+    // bound listeners
+    onEnterListener: (event: MouseEvent) => void;
+    onLeaveListener: (event: MouseEvent) => void;
```

Comment:
These always struck me as odd. Who is the caller?

Reply:
I'm not sure what you mean? These are for binding callbacks to `this` and saving the result so we can remove them later on.